src/Indice.Features.Cases.AspNetCore/Services/RoleCaseTypeService.cs

```diff
@@ -77,49 +85,79 @@ IDistributedCache distributedCache
                 .Any(x => x.CaseTypeId == caseDetails.CaseType!.Id && x.CheckpointTypeId == caseDetails.CheckpointTypeId);
         }
 
+        /// <summary>
+        /// Determines whether user is Owner of a Case
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="caseDetails"></param>
         private bool IsOwnerOfCase(ClaimsPrincipal user, CaseDetails caseDetails) {
             return user.FindSubjectId().Equals(caseDetails.CreatedById);
         }
 
+        /// <summary>
+        /// Gets the list of allowed and filtered CheckpointType Ids for admin users
+        /// </summary>
+        /// <param name="checkpointTypeCodes"></param>
         private async Task<List<string>> ApplyAdminCheckpointTypeFilter(List<string> checkpointTypeCodes) {
-            if (checkpointTypeCodes == null || checkpointTypeCodes.Count == 0) {
-                return checkpointTypeCodes;
+            // admin didn't select checkpointTypeCode filters -> get all checkpointTypeIds
+            if (checkpointTypeCodes is null || checkpointTypeCodes.Count == 0) {
+                return await _dbContext.CheckpointTypes
+                .AsQueryable()
+                .Select(checkpointType => checkpointType.Id.ToString())
+                .ToListAsync();
             }
-            var codes = await _dbContext.CheckpointTypes
+            // admin selected checkpointTypeCode filters
+            return await _dbContext.CheckpointTypes
                 .AsQueryable()
-                .Select(c => c.Code)
-                .AsAsyncEnumerable()
-                .Distinct() // TODO client-side evaluation, this needs to change
```

Comment:
The method changed completely. Is this comment still relevant?

Reply:
Σωστά δεν ισχύει εδώ, αλλά ίσως ισχύει (θέλει sql profiler ή debug να δουμε τι query φτανει εν τελει) το ίδιο για την γραμμή 128 `.Distinct() // Avoid duplicates: it is possible that user has >1 roles and those roles can "see" common CheckpointTypes`